metrics-scraper/monitoring.go

```diff
@@ -159,33 +226,48 @@ func nodeVersion(ctx context.Context, client *rpc.Client, moniker, networkID str
 	return nil
 }
 
-func sidecarsVersion(ctx context.Context, client *rpc.Client, moniker, networkID string) error {
+func sidecarsVersion(ctx context.Context, client *rpc.Client, moniker, networkID string) (err error) {
+	defer func() {
+		if err != nil {
+			ethereumSidecarGauge.WithLabelValues(moniker, networkID, "0.0.0-unknown").Set(0)
+			connectSidecarGauge.WithLabelValues(moniker, networkID, "0.0.0-unknown").Set(0)
+		}
+	}()
+
+	// always remove the previous metrics to ensure that there's no duplicates when
+	// the labels version changes
+	ethereumSidecarGauge.DeletePartialMatch(map[string]string{
+		"moniker":  moniker,
+		"chain_id": networkID,
+	})
+	connectSidecarGauge.DeletePartialMatch(map[string]string{
+		"moniker":  moniker,
+		"chain_id": networkID,
+	})
+
 	result := map[string]net.SidecarInfos{}
-	err := client.CallContext(ctx, &result, netSidecarsEndpoint)
+	err = client.CallContext(ctx, &result, netSidecarsEndpoint)
 	if err != nil {
+		// just format nicely the error, and continue execution
+		// so the prometheus variable are still set to defaults
 		return fmt.Errorf("couldn't call %v for %v: %v", netSidecarsEndpoint, moniker, err)
 	}
 
-	var (
-		ethereumVersion, ethereumIsRunning = "unknow", 0.
-		connectVersion, connectIsRunning   = "unknow", 0.
-	)
-
 	if ethereumSidecar, ok := result["ethereum"]; ok {
+		var isConnected float64
 		if ethereumSidecar.Connected {
-			ethereumIsRunning = 1
+			isConnected = 1
 		}
-		ethereumVersion = ethereumSidecar.Version
+		ethereumSidecarGauge.WithLabelValues(moniker, networkID, ethereumSidecar.Version).Set(isConnected)
 	}
-	ethereumSidecarGauge.WithLabelValues(moniker, networkID, ethereumVersion).Set(ethereumIsRunning)
 
 	if connectSidecar, ok := result["connect"]; ok {
+		var isConnected float64
 		if connectSidecar.Connected {
-			connectIsRunning = 1
+			isConnected = 1
 		}
-		connectVersion = connectSidecar.Version
+		connectSidecarGauge.WithLabelValues(moniker, networkID, connectSidecar.Version).Set(isConnected)
 	}
-	connectSidecarGauge.WithLabelValues(moniker, networkID, connectVersion).Set(connectIsRunning)
 
-	return nil
+	return err
```

Comment:
```suggestion
	return
```

Reply:
addressed in: 94e0597d570fbb1a694bc047ed7b89d8570a1289